### PR TITLE
fix(client): use Gatsby Link for internal links

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

During the Bootstrap, we migrated most of our button links (links that look like a button) to the `@freecodecamp/ui` Button component. The Button component renders a link if it receives an `href`. However, the link it renders is a plain `<a>`, which doesn't give us the [performance benefits](https://v3.gatsbyjs.com/docs/tutorial/part-2/#use-the-link-component) Gatsby offers, and also causes navigation issues on non-English /learn (the links will take the users to the English pages).

This PR:
- Creates a ButtonLink component that:
  - Renders a Gatsby Link if the link is internal
  - Renders a @freecodecamp/ui Button if the link is external
- Migrates a couple of Button to ButtonLink (@sidemt found navigation issues in #55019, so I'm fixing those first and will raise a follow-up PR to migrate the rest)

Note: Reviewing by commit is recommeded.

---
This isn't related to the PR, but I'm just dumping my thoughts here.

We will need to rethink how we want to build our link component in `@freecodecamp/ui`. Right now the library can only provide a component for external links, and its consumers (both current and potential) don't use the same UI framework:
- /learn uses Gatsby
- /contribute uses Astro
- CAP - I'm assuming we aren't going to use Gatsby here

We would either need to add all sorts of link components to the library, or provide styling utilities from the library:

```tsx
// @freecodecamp/ui
export const useLinkStyles = () => {
  // return a bunch of Tailwind classes
}

// Client
import { GatsbyLink } from 'gatsby'
import { useLinkStyles } from '@freecodecamp/ui'

const Link = () => {
  const cls = useLinkStyles()
  return <GatsbyLink className={cls} />
}
```

### CODE REVIEW

**Review by ahmaxed — APPROVED** [review]

File: client/src/components/helpers/button-link.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    } else if (size === 'small') {
+      cls.push('btn-sm');
+    }
+
+    if (className) {
+      cls.push(className);
+    }
+
+    return cls.join(' ');
+  }, [size, block, className]);
```
Comment (ojeytonwilliams) [inline client/src/components/helpers/button-link.tsx:60]:
```suggestion
  const cls = ['btn'];
  if (block) cls.push('btn-block');

  // The 'btn' class contains the base button styles as well as the `medium` variant styles.
  // So we only need to handle `large` and `small`.
  if (size === 'large') cls.push('btn-lg');
  if (size === 'small') cls.push('btn-sm');
  if (className) cls.push(className);

  const gatsbyLinkCls = cls.join(' ');
```
Is a memo useful here? None of this is expensive. Also, Naomi would disapprove, but dropping the `{}` cleans this up a ton.

File: client/src/components/helpers/button-link.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        className={className}
+        size={size}
+        block={block}
+        disabled
+      >
+        {children}
+      </Button>
+    );
+  }
+
```
Comment (ojeytonwilliams) [inline client/src/components/helpers/button-link.tsx:76]:
```suggestion
```
Can't we pass all the props to the Button and let it deal with it?

File: client/src/components/helpers/button-link.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  if (target === '_blank' || download) {
+    return (
+      <Button
+        variant='primary'
+        className={className}
```
Comment (ojeytonwilliams) [inline client/src/components/helpers/button-link.tsx:81]:
```suggestion
        className={className}
        disabled
```
See above.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
Looking good. Nice job, @huyenltnguyen

**Pull request merged**